File: src/main/scala/im/tox/toktok/app/call/CallMessageAdapter.scala
package im.tox.toktok.app.call

import android.support.v7.widget.RecyclerView
import android.view.{ LayoutInflater, ViewGroup }
import android.widget.{ LinearLayout, TextView }
import im.tox.toktok.TR
import im.tox.toktok.TypedResource._

final class CallMessageAdapter(excuses: java.util.List[String]) extends RecyclerView.Adapter[CallMessageViewHolder] {

  def onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): CallMessageViewHolder = {
    val itemView = LayoutInflater.from(viewGroup.getContext).inflate(TR.layout.call_item, viewGroup, false)
    new CallMessageViewHolder(itemView)
  }

  def onBindViewHolder(viewHolder: CallMessageViewHolder, position: Int): Unit = {
    val item = excuses.get(position)
    viewHolder.mMessage.setText(item)
  }

  def getItemCount: Int = {
    excuses.size()
  }

}

final class CallMessageViewHolder(itemView: LinearLayout) extends RecyclerView.ViewHolder(itemView) {

  val mMessage: TextView = itemView.findView(TR.call_item_message)

}

File: sources/app/src/main/java/im/tox/toktok/app/call/CallMessageAdapter.java
package im.tox.toktok.app.call;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import im.tox.toktok.R;

final class CallMessageAdapter extends RecyclerView.Adapter<CallMessageViewHolder> {

    private final List<String> excuses;

    CallMessageAdapter(List<String> excuses) {
        this.excuses = excuses;
    }

    @NonNull
    @Override
    public CallMessageViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        LinearLayout itemView = (LinearLayout) LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.call_item, viewGroup, false);
        return new CallMessageViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull CallMessageViewHolder viewHolder, int position) {
        String item = excuses.get(position);
        viewHolder.mMessage.setText(item);
    }

    @Override
    public int getItemCount() {
        return excuses.size();
    }

}

final class CallMessageViewHolder extends RecyclerView.ViewHolder {

    final TextView mMessage;

    CallMessageViewHolder(@NonNull LinearLayout itemView) {
        super(itemView);
        this.mMessage = itemView.findViewById(R.id.call_item_message);
    }

}
